docker-compose.yml

```diff
@@ -14,6 +14,13 @@ services:
       spring.datasource.password: admin
       spring.datasource.driverClassName: org.postgresql.Driver
       spring.jpa.properties.hibernate.dialect: ca.uhn.fhir.jpa.model.dialect.HapiFhirPostgres94Dialect
+      OAUTH_ENABLED: true
+      OAUTH_ISSUER: https://auth-internal.elimuinformatics.com/auth/realms/product
+      OAUTH_JWKS_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/certs
+      OAUTH_AUTHORIZATION_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/auth
+      OAUTH_TOKEN_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/token
+      # OAUTH_INTROSPECTION_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/token/introspect
+      # OAUTH_REVOCATION_URL: https://auth-internal.elimuinformatics.com/auth/realms/product/protocol/openid-connect/revoke
```

Comment:
Are these OAUTH properties used for anything other than .well-known/smart-configuration? If we enable this on fhir2-internal, is it going to enable authorization so we cannot access the server without a token? Or can we still serve a smart-configuration with OAuth disabled?

Reply:
So that means we need to enable OAuth for fhir2-internal in order to serve a .well-known/smart-configuration? We have a lot of integration tests, postman collections, etc. that rely on fhir2-internal being open. They would all need to be updated to add oauth flow.

Also, if we want to start using this right away (so we can make smart-launcher private), we will need to list smart-launcher as the auth server in smart-configuration, and I don't know if it makes sense to use the smart-launcher as the OAuth server as well.

If we are going to adopt authorization incrementally, we may want separate configuration for OAuth and SMART

Reply:
No, the .well-known/smart-configuration will be served regardless. Sorry I wasn't clear. The OAuth2 authorization enforcement only occurs if OAUTH_ENABLED is true.